tests/models/sam_hq/test_modeling_sam_hq.py

```diff
@@ -0,0 +1,863 @@
+        processor = SamHQProcessor.from_pretrained("sushmanth/sam_hq_vit_b")
+
+        model.to(torch_device)
+        model.eval()
+
+        raw_image = prepare_image()
+        input_boxes = [[[620, 900, 1000, 1255]]]
+        input_points = [[[820, 1080]]]
+        labels = [[0]]
+
+        inputs = processor(
+            images=raw_image,
+            input_boxes=input_boxes,
+            input_points=input_points,
+            input_labels=labels,
+            return_tensors="pt",
+        ).to(torch_device)
+
+        with torch.no_grad():
+            outputs = model(**inputs)
+        scores = outputs.iou_scores.squeeze()
+
+        self.assertTrue(torch.allclose(scores[-1], torch.tensor(0.8680), atol=1e-4))
+
+    def test_inference_mask_generation_one_point(self):
+        model = SamHQModel.from_pretrained("sushmanth/sam_hq_vit_b")
+        processor = SamHQProcessor.from_pretrained("sushmanth/sam_hq_vit_b")
+
+        model.to(torch_device)
+        model.eval()
+
+        raw_image = prepare_image()
+
+        input_points = [[[400, 650]]]
+        input_labels = [[1]]
+
+        inputs = processor(
+            images=raw_image, input_points=input_points, input_labels=input_labels, return_tensors="pt"
+        ).to(torch_device)
+
+        with torch.no_grad():
+            outputs = model(**inputs)
+        scores = outputs.iou_scores.squeeze()
+        self.assertTrue(torch.allclose(scores[-1], torch.tensor(0.9137), atol=1e-4))
+
+        # With no label
+        input_points = [[[400, 650]]]
+
+        inputs = processor(images=raw_image, input_points=input_points, return_tensors="pt").to(torch_device)
+
+        with torch.no_grad():
+            outputs = model(**inputs)
+        scores = outputs.iou_scores.squeeze()
+        self.assertTrue(torch.allclose(scores[-1], torch.tensor(0.9137), atol=1e-4))
+
+    def test_inference_mask_generation_two_points(self):
+        model = SamHQModel.from_pretrained("sushmanth/sam_hq_vit_b")
+        processor = SamHQProcessor.from_pretrained("sushmanth/sam_hq_vit_b")
+
+        model.to(torch_device)
+        model.eval()
+
+        raw_image = prepare_image()
+
+        input_points = [[[400, 650], [800, 650]]]
+        input_labels = [[1, 1]]
+
+        inputs = processor(
+            images=raw_image, input_points=input_points, input_labels=input_labels, return_tensors="pt"
+        ).to(torch_device)
+
+        with torch.no_grad():
+            outputs = model(**inputs)
+        scores = outputs.iou_scores.squeeze()
+        self.assertTrue(torch.allclose(scores[-1], torch.tensor(0.8859), atol=1e-4))
+
+        # no labels
+        inputs = processor(images=raw_image, input_points=input_points, return_tensors="pt").to(torch_device)
+
+        with torch.no_grad():
+            outputs = model(**inputs)
+        scores = outputs.iou_scores.squeeze()
+
+        self.assertTrue(torch.allclose(scores[-1], torch.tensor(0.8859), atol=1e-4))
```

Comment:
same, precision varies a bit
```suggestion
        self.assertTrue(torch.allclose(scores[-1], torch.tensor(0.8859), atol=1e-3))
```

Reply:
done

Reply:
seems it's back to 1e-4 here?

Reply:
Can we bring this back to 1e-3 please? tests will break otherwise. @sushmanthreddy (the thread might have been buried by github)

Reply:
```suggestion
        self.assertTrue(torch.allclose(scores[-1], torch.tensor(0.8859), atol=1e-3))
```